alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -494,6 +541,31 @@ class TransportIndexMonitorAction @Inject constructor(
             if (request.monitor.enabled && currentMonitor.enabled)
                 request.monitor = request.monitor.copy(enabledTime = currentMonitor.enabledTime)
 
+            // On update monitor check which backend roles to associate to the monitor
+            if (user != null) {
+                if (request.rbacRoles != null) {
+                    if (isAdmin(user)) {
+                        request.monitor = request.monitor.copy(
+                            user = User(user.name, request.rbacRoles, user.roles, user.customAttNames)
+                        )
+                    } else {
+                        val rolesToRemove = user.backendRoles
+                        request.rbacRoles.orEmpty().let { rolesToRemove.removeAll(it) }
+                        val rolesToAdd = request.rbacRoles?.intersect(user.backendRoles)
```

Comment:
Is it possible for there to be roles that are in `request.rbacRoles` that aren't in `user.backendRoles`? We do a validation early on that throws an error if the request has backend roles the user doesn't have permissions to. In that case, could we just use `request.rbacRoles`?

Reply:
cleaned it up